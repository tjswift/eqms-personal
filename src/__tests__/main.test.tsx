import React from 'react';
import Main from '../main';
import { createBrowserHistory } from 'history'
import configureStore from '../configureStore'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('The main component', () => {
    it('renders without crashing', () => {
        const history = createBrowserHistory()

        const initialState = window.initialReduxState
        const store = configureStore(history, initialState)
        shallow(<Main store={store} history={history} />);
    });

});


describe('Hello, Enzyme!', () => {
    it('renders', () => {
        const wrapper = shallow(<div>
            <h1>Hello, Enzyme!</h1>
        </div>)
        expect(wrapper.find('h1').html()).toMatch(/Hello, Enzyme/)
    });

    it('renders snapshots, too', () => {
        const wrapper = shallow(<div>
            <h1>Hello, Enzyme!</h1>
        </div>)
        expect(wrapper).toMatchSnapshot()
    });
})