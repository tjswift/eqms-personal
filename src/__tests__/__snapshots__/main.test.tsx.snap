// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello, Enzyme! renders snapshots, too 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <h1>
      Hello, Enzyme!
    </h1>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Hello, Enzyme!
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello, Enzyme!",
      },
      "ref": null,
      "rendered": "Hello, Enzyme!",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hello, Enzyme!
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello, Enzyme!",
        },
        "ref": null,
        "rendered": "Hello, Enzyme!",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
